feat: expose some configuration options to home manager module

snaeil opened the pull request:
This is currently more an RFC @adriankarlen and maybe @shadeyg56 .

Since [custom configuration](https://github.com/adriankarlen/textfox/pull/53) is possible, it's easy to adapt the nix home-manager module.

With this PR, it would be possible to configure font family, font size and horizontal tabs, using nix.

**Advantages:**
- Allowing a (nix-)idiomatic usage of textfox
- Much more convenient for home manager users

**Potential challenge:**
- Changes in defaults always have to be taken into account for `hm-module.nix` to preserve the documented behaviour

If you would be willing to merge such a feature @adriankarlen , I can finalize the missing options.
I just don't yet understand all  config options, e.g. `--tf-accent`. I think it would be a good idea to explain these in the README's custom config section,

e.g.

```css
  --tf-rounding: 0px; /*the border radius*/
```

adriankarlen commented:
Yeah, good point regarding the variables, and what their usage is. This sounds like a really cool change, will update the documentation.

adriankarlen commented:
Documentation has now been improved in #61, hopefully this helps you with this PR.

snaeil commented:
Thanks, I am currently working on this

snaeil commented:
@adriankarlen @shadeyg56 I cleaned up my work to provide it as one commit.
It also includes documentation now.

Since I am still quite new to nix, there might be things I am not aware of, that can be done better and refactored.
Please let me know if would like me to change something.

shadeyg56 commented:
LGTM! I've been using Nix for well over a year and still am not very good at packaging software haha.

Thanks for helping to improve Nix support

adriankarlen commented:
Thanks for hel ping out with the review @shadeyg56 . I'll merge this then.
